### DISCUSSION TITLE
How to map to get lower than or greater than symbols ?

### DISCUSSION THREAD

**Discussion by jeanpat** [sezanzeb/input-remapper#215] [Q&A]
Hello,
I have an arch-based linux (endeavourOS) installed on a thinkpad T430 which have some kind of azerty keyboard:

```
PS/2 00.0: 10800 Keyboard
  [Created at input.226]
  Unique ID: 9ui9.+49ps10DtUF
  Hardware Class: keyboard
  Model: "AT Translated Set 2 keyboard"
  Vendor: 0x0001 
  Device: 0x0001 "AT Translated Set 2 keyboard"
  Compatible to: int 0x0211 0x0001
  Device File: /dev/input/event3
  Device Files: /dev/input/event3, /dev/input/by-path/platform-i8042-serio-0-event-kbd
  Device Number: char 13:67
  Driver Info #0:
    XkbRules: xfree86
    XkbModel: pc104
  Config Status: cfg=new, avail=yes, need=no, active=unknown
```

Unfortunately, I'm afraid that no key combinations allow to type < or >. I guess it should be possible to use key-mapper to fix this problem. I've tried:

![key-mapper](https://user-images.githubusercontent.com/693574/143619198-2957c125-c5a4-485e-b4b2-33de273cd928.png)

Thanks for advices

**Comment by sezanzeb** [comment]
You can try this guide at the bottom: https://github.com/sezanzeb/key-mapper/blob/main/readme/examples.md#emitting-unavailable-symbols, but using xmodmap alone without key-mapper might already be able to do the trick.

Since key-mapper is acting like a keyboard it has the same constraints of usb keyboards. The keyboard layouts defines which symbols are possible to be written.

Have you tried xmodmap? But I think it only works in x11, you can find out what you are on using `echo $XDG_SESSION_TYPE`.

**Reply by jeanpat** [reply]
EndeavourOS uses wayland, is it possible to use xmodmap?
$ echo $XDG_SESSION_TYPE
wayland

**Reply by sezanzeb** [reply]
As far as I know your only option is to write xkb configs, which is a huge pain. Actually there is a branch of key-mapper that generates xkb configs for you, but it didn't work in one case.

I just merged in all new changes and made sure all tests pass and tried it out, for me it works: https://github.com/sezanzeb/key-mapper/tree/xkb

![image](https://user-images.githubusercontent.com/28510156/143678905-35004e1a-144b-45e1-950e-6bc7a4cf210d.png)

all possible names can be found in: https://github.com/sezanzeb/key-mapper/blob/xkb/keymapper/valid_symbols.py

But I don't know if it works in wayland, since the environment has to care about the setxkbmap command

Actually, I think this is amazing and it was a lot of work to get it to this point. But I need some more feedback from people if it works

**Reply by sezanzeb** [reply]
Here is what it does in a nutshell:

```
154353 12:14:00.466139 INFO xkb.py:159: Unknown characters found, creating xkb configs
154353 12:14:00.466367 SPAM xkb.py:120: "greaterthanequal" (84) is allocated
154353 12:14:00.466451 INFO paths.py:50: Creating file "/usr/share/X11/xkb/symbols/key-mapper/Razer_Razer_Naga_Trinity"
154353 12:14:00.466947 INFO xkb.py:169: Writing xkb symbols "/usr/share/X11/xkb/symbols/key-mapper/Razer_Razer_Naga_Trinity"
154353 12:14:00.466998 SPAM xkb.py:171: "/usr/share/X11/xkb/symbols/key-mapper/Razer_Razer_Naga_Trinity":
default xkb_symbols "key-mapper" {
    key <92> { [ greaterthanequal ] };
};
154293 12:14:00.969837 INFO xkb.py:96: Applying xkb configuration
154293 12:14:00.974979 DEBUG xkb.py:112: Running "setxkbmap -keycodes key-mapper-keycodes -symbols key-mapper/Razer_Razer_Naga_Trinity -device 15"
```

**Reply by sezanzeb** [reply]
Looks like there will be difficulties in applying the mapping: https://itectec.com/unixlinux/does-wayland-use-xkb-for-keyboard-layouts/

And adding one alternative method to the setxkbmap command for each possible environment defies my philosophy about key-mapper.

**Reply by sezanzeb** [reply]
And this only works if the distro provides a useful `xmodmap -pke` output, which I think isn't the case in gnome.

All in all, in wayland your options are incredibly limited/complicated right now

**Comment by w-nrt** [comment]
Hello people with the same issue! "shift_l + less" (on AZERTY) returns >
On QWERTY "KEY_LEFTSHIFT + KEY_DOT" output is >
